[feat] Add agent authentication types reference (#26)
libremfg/libremfg.github.io: content/reference/agent-authentication.md

---
title: 'Agent authentication types'
categories: ["reference"]
description: Authentication types for the Rhize agent
weight: 900
menu:
  main:
    parent: reference
    identifier:
---

> The Rhize agent collects data emitted that is emitted in the manufacturing process and makes this data visible in the Rhize system.

Reviewer: > 
```suggestion
The Rhize agent collects data that is emitted in the manufacturing process and makes this data visible in the Rhize system.
```

Just a double word.

> The Rhize agent collects data emitted that is emitted in the manufacturing process and makes this data visible in the Rhize system.

Author: ```suggestion
The Rhize agent collects data that is emitted in the manufacturing process and makes this data visible in the Rhize system.
```

It works by connecting to equipment or groups of equipment that run over protocols such as OPC UA,

> and then reading and writing tags to re-publish them to NATS.

Reviewer: ```suggestion
and then subscribing to tags to re-publish them to NATS.
```

Then I'd add something about also providing datasource reads/writes and method calls via BPMN.

Author: [185a9c8](https://github.com/libremfg/libremfg.github.io/pull/26/commits/185a9c8ddb1d7b7118b7761608a5d2da6bc6cab5) Expands the intro a bit

> As the communication bridge between the Rhize Data Hub and your plant, the agent can handle reads and writes to and from data sources, and it can be called by a [BPMN task]({{< relref "bpmn/activities" >}}).
> 
> It works by connecting to equipment or groups of equipment that run over protocols such as OPC UA,
> and then reading and writing tags to re-publish them to NATS.

Author: ```suggestion
It works by connecting to equipment or groups of equipment that run over protocols such as OPC UA.

As the communication bridge between the Rhize Data Hub and your plant, the agent has multiple functions:
- It subscribes to tags and republishes the changes in NATS.
- It also exposes read and write commands for BPMN to send reads and writes to equipment.
```

## OPC UA authentication types

 When authenticating over OPC UA, Rhize supports the following authentication types:

| Authentication type | Behavior                                                                                                                                                                                       |
|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Anonymous           | Connects without any necessary credential                                                                                                                                                      |

> | Username            | Authenticates through a `username` and `password` in the config file                                                                                                                           |

Reviewer: Or (eventually) kubernetes secret.

Author: ```suggestion
| Username            | Authenticates through a `username` and `password` in the config file, or through a Kubernetes secret                                                                                                                           |
```

> | Certificate         | Uses the certificate on disk specified in the config at `OPCUA.CertFile/KeyFile`. If none exists and the config specifies the `OPCUA.GenCert` property as `true`, automatically generates one. |

Reviewer: These are two individual configs. `OPCUA.CertFile` and `OPCUA.KeyFile`.

Author: Would the user need to configure both? Or is this an either/or situation?

Reviewer: Both, a key isn't useful unless it's signed by a CA, which is done in a certificate. On the other side, the certificate is public, so it's useless without the private key.

> | Certificate         | Uses the certificate on disk specified in the config at `OPCUA.CertFile/KeyFile`. If none exists and the config specifies the `OPCUA.GenCert` property as `true`, automatically generates one. |

Author: ```suggestion
| Certificate         | Uses the certificate on disk specified in the  `OPCUA.CertFile` and `OPCUA.KeyFile` configs. If no certificate exists and the config specifies the `OPCUA.GenCert` property as `true`, automatically generates one. |
```
